test/tuple.jl

```diff
@@ -845,3 +845,5 @@ end
         end
     end
 end
+
+@test NTuple == Core.Compiler.return_type(reverse, Tuple{NTuple})
```

Comment:
```suggestion
@test NTuple == Base.infer_return_type(reverse, Tuple{NTuple})
```
It would be better to use `Base.infer_return_type` for inference accuracy test in general.